subiquity/server/controllers/filesystem.py

```diff
@@ -127,23 +126,39 @@
 DRY_RUN_RESET_SIZE = 500 * MiB
 
 
-class NonReportableSVE(RecoverableError, NonReportableException):
-    """Non reportable storage value error"""
+class StorageRecoverableError(RecoverableError):
+    pass
 
 
-class ReportableSVE(ValueError):
-    """Reportable storage value error"""
+class StorageInvalidUsageError(StorageRecoverableError):
+    """Exception to raise when a storage endpoint is not properly used."""
 
-    # TODO Inherit from RecoverableError going forward.
+    code = "storage-invalid-usage"
+    title = _("Invalid use of storage endpoint")
 
 
-# Depending on config, we will let the SVE fail on the server side
-StorageRecoverableError: Type[NonReportableSVE | ReportableSVE] = ReportableSVE
+class StorageConstraintViolationError(StorageRecoverableError):
```

Comment:
praise: these subclass definitions read really nice. 

At first I had wondered why you didn't use the usual abstract base class machinery in `RecoverableError` and use properties to get around the weird init, but this way definitely reads nicer I think.